aiida/orm/entities.py

```diff
@@ -163,6 +163,16 @@ def all(self):
         """
         return [_[0] for _ in self.query().all()]
 
+    @classmethod
+    def reset(cls):
+        """
+        Reset collection to initial state.
+
+        Resets collection instance back to its initial state (to be constructed from scratch when next requested).
+        Useful to clear any internal caches of the collection.
+        """
+        cls._COLLECTIONS = datastructures.LazyStore()
```

Comment:
I don't think this does what you intend it to do. This will reset the collection for _all_ entities and backend. So when you call `User.object.reset` it will reset all collections not just that of `User`. I would really go for the alternative you proposed for now and implement a specific default user reset on the `User` collection

Reply:
Thanks, for the moment I've implemented the modification described in #3462 

I can still change it if this is thought to be the worse solution.